libc/src/__support/float_to_string.h

```diff
@@ -488,7 +489,8 @@ class FloatToString {
 
       val = POW10_SPLIT[POW10_OFFSET[idx] + block_index];
 #endif
-      const uint32_t shift_amount = SHIFT_CONST + (IDX_SIZE * idx) - exponent;
+      const uint32_t shift_amount =
+          static_cast<uint32_t>(SHIFT_CONST + (IDX_SIZE * idx) - exponent);
```

Comment:
I think just IDX_SIZE is a size_t; You could limit the static cast to just those (or perhaps define IDX_SIZE as a smaller type?)